File: InterfaceBuilding/SwiftInterfaceBulding/SwiftInterfaceBulding/FullRebuildTester.swift
//
//  FullRebuildTester.swift
//  SwiftInterfaceBulding
//
//  Created by Krzysztof Drab on 29/12/2020.
//  Copyright © 2020 Krzysztof Drab. All rights reserved.
//

import Foundation

class FullRebuildTester: Tester {
    private var nbOfItems: Int = 0
    private var rows: [RowData] = []
    
    override func generateRows() -> [RowData] {
        self.tests = [1, 250, 500, 750, 1000, 1250, 1500, 1750, 2000]
        self.nbOfItems = tests[testId];
        self.rows = self.generateDefaultRows(nbOfRows: self.nbOfItems);
        return self.rows;
    }

    override func updateRows() -> [RowData] {
        let startingPoint = Date()
        var i = 0;
        while(i < self.rows.count) {
            self.rows[i] = RowData(id: Int.random(in: 1...10000), value: Double.random(in: 0...1))
            i += 1
        }
        self.timeToModify += startingPoint.timeIntervalSinceNow * -1;
        return self.rows;
    }
    
    override func testEnded(fps: Double) {
        self.results.append(TestResult(n: nbOfItems, timeToModify: timeToModify, fps: fps))
    }
}

File: InterfaceBuilding/flutter_interface_building/lib/full_rebuild_tester.dart
import 'dart:io' show Platform;

import 'row_data.dart';
import 'tester.dart';

class FullRebuildTester extends Tester {
  List<int> tests = Platform.isIOS
    ? [1, 250, 500, 750, 1000, 1250, 1500, 1750, 2000]
    : [1, 50, 100, 150, 200, 250, 300, 350, 400];
  int nbOfItems;
  List<RowData> rows;

  List<RowData> generateRows() {
    this.nbOfItems = this.tests[this.testId];
    this.rows = this.generateDefaultRows(this.nbOfItems);
    return this.rows;
  }

  List<RowData> updateRows() {
    final stopwatch = Stopwatch()..start();
    this.rows = this.generateDefaultRows(this.nbOfItems);
    this.timeToModify += stopwatch.elapsedMilliseconds;
    return this.rows;
  }

  void testEnded(double fps) {
    this.results.add(TestResult(this.nbOfItems, fps, this.timeToModify));
  }
}